src/main/java/org/jabref/logic/journals/PredatoryJournalLoader.java

```diff
@@ -0,0 +1,145 @@
+package org.jabref.logic.journals;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.Reader;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardCopyOption;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.jabref.logic.net.URLDownload;
+import org.jabref.logic.util.OS;
+
+import net.harawata.appdirs.AppDirsFactory;
+import org.apache.commons.csv.CSVFormat;
+import org.apache.commons.csv.CSVParser;
+import org.apache.commons.csv.CSVRecord;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class PredatoryJournalLoader {
```

Comment:
The loader should only be used in `PredatoryJournalListMvGenerator.java`

Reply:
The class behaves now similar to the JournalAbbreviationLoader. The Loader loads the repository from file, e.g. the mv database. It has a method for updating the predatory journals. However, that is only called from the Generator.